File: Backend/Models/cliente.py
from .data_base import db

class Cliente(db.Model):

    __tablename__ = 'clientes'

    id = db.Column(db.Integer, primary_key=True)
    nome = db.Column(db.String(100), nullable=False)
    cpf = db.Column(db.String(14), unique=True, nullable=False)
    email = db.Column(db.String(100), unique=True, nullable=False)
    # Werkzeug gera hashes maiores que 100 chars; senão o MySQL quebra com erro de tamanho.
    senha = db.Column(db.String(255), nullable=False)
    celular = db.Column(db.String(50), nullable=False)
    cep = db.Column(db.String(9), nullable=False)
    endereco = db.Column(db.String(100), nullable=False)
    numero = db.Column(db.Integer, nullable=False)
    complemento = db.Column(db.String(100), nullable=True)

    def para_dicionario(self):
        dados = {
            'id': self.id,
            'nome': self.nome,
            'cpf': self.cpf,
            'email': self.email,
            'celular': self.celular,
            'cep': self.cep,
            'endereco' : self.endereco,
            'numero' : self.numero,
            'complemento' : self.complemento
        }
        return dados

File: Frontend/JS/cliente.js
function validar_dados() {
    let nome = document.getElementById("nome").value;
    let email = document.getElementById("email").value;
    let cpf = document.getElementById("cpf").value;
    let senha = document.getElementById("senha").value;
    let celular = document.getElementById("celular").value;
    let cep = document.getElementById("cep").value;
    let endereco = document.getElementById("endereco").value;
    let numero = document.getElementById("numero").value;

    if (!nome || !email || !cpf || !senha || !celular || !cep || !endereco || !numero) {
        alert("Preencha todos os campos!");
        return;
    }

    enviar_cadastro(nome, email, cpf, senha, celular, cep, endereco, numero);
}

async function enviar_cadastro(nome, email, cpf, senha, celular, cep, endereco, numero) {
    let complemento = document.getElementById("complemento").value;

    let dados = {
        nome: nome,
        cpf: cpf,
        email: email,
        senha: senha,
        celular: celular,
        cep: cep,
        endereco: endereco,
        numero: numero,
        complemento: complemento
    };

    try {
        let response = await fetch("http://localhost:5000/clientes", {
            method: "POST",
            body: JSON.stringify(dados),
            headers: {
                "Content-Type": "application/json"
            }
        });

        let resposta = await response.json();

        if (!response.ok) {
            throw new Error(resposta.mensagem || "Erro ao cadastrar cliente");
        }

        alert("Cadastro realizado com sucesso!");
        window.location.href = "../pages/login.html";

    } catch (error) {
        console.error("Erro:", error);
        alert(error.message);
    }
}

async function autenticar_login() {
    let email = document.getElementById("email").value;
    let senha = document.getElementById("senha").value;

    if (!email || !senha) {
        alert("Preencha email e senha!");
        return;
    }

    let dados = {
        email: email,
        senha: senha
    };

    try {
        let request = await fetch("http://localhost:5000/clienteslogin", {
            method: "POST",
            body: JSON.stringify(dados),
            headers: {
                "Content-Type": "application/json"
            }
        });

        let resposta = await request.json();

        if (!request.ok) {
            throw new Error(resposta.mensagem || "Email ou senha inválidos");
        }

        alert("Login realizado com sucesso!");
        window.location.href = "../pages/dashboard.html";

    } catch (error) {
        console.error("Erro:", error);
        alert(error.message);
    }
}